cli/tools/lint/linter.rs

```diff
@@ -257,3 +266,105 @@ fn apply_lint_fixes(
     deno_ast::apply_text_changes(text_info.text_str(), quick_fixes);
   Some(new_text)
 }
+
+#[derive(Debug, Default)]
+pub struct FileMetrics {
+  avg_line_length: f64,
+  long_line_percentage: f64,
+  whitespace_ratio: f64,
+  has_source_map: bool,
+  short_var_name_ratio: f64,
+}
+
+pub fn is_minified_file(code: &str) -> bool {
+  let mut metrics = FileMetrics::default();
+
+  // Split into non-empty lines
+  let lines: Vec<&str> = code
+    .lines()
+    .map(str::trim)
+    .filter(|line| !line.is_empty())
+    .collect();
```

Comment:
Seems like a lot of work and allocations. Maybe we should just do something that stops after X chars on the first line? (ex. read the first 400 chars and if there's no newline then it's probably minified)